### ISSUE TITLE
Can I ask you how you train the base model?

### ISSUE THREAD

**Issue by stephenjia** [aimerykong/Recurrent-Scene-Parsing-with-Perspective-Understanding-in-the-loop#7]
Hi Shu,

Can I ask you how you train the base model? What are the hyperparameters for training from deeplab resnet model file to get a good base model as mentioned in the paper?
I check the given checkpoint and find that you set learning rate to 0 for layers in base model and also pyramidpooling layers. 

Thanks a lot!
Best,
Xu

**Comment by aimerykong** [comment]
Hi, Xu,
For the model you see, I use pspnet as a part of base model. Hope this helps! 

Regards,
Shu

**Comment by stephenjia** [comment]
Hi Shu,

Thank you very much for the quick reply. I am not sure whether my understanding is correct. Does that mean in base model you use the same ResNet backbone as in PSPNet, and only update the parameters of two newly added conv layers but not the ones of ResNet backbone? I find in the model file learning rates of the two newly added conv layers are also set to 0.
I have another question. I notice that the two newly added conv layers are also followed by batch normalization and relu layers. How do you get the parameters and statistics for these two bn layers, especially when batch size is set to 1?

Thanks a lot.

Best,
Xu

**Comment by aimerykong** [comment]
hi, Xu,

As for the model I release, I copy the resnet-like backbone of pspnet, and
build new layers which are randomly initialized. Of course, other
pre-trained backbones can also be used like DeepLab. In the new layers, I
also build batch normalization. To train it, I adopt a stage-wise
procedure. First, I freeze the backbone and only train new layers. It's
interesting that even I set batch size as one, feeding one image, I have no
problem in training the batch normalization layer. Once the new layers are
trained to a reasonable point, I include the backbone and fine-tune. The
stage-wise training procedure is also adopted when training the depth-aware
gating module and recurrent module.

If you have further questions and findings, I'm happy to discuss.

Regards,
Shu

**Comment by stephenjia** [comment]
Hi Shu,

Thank you so much for your detailed reply. Now I understand better how you trained the model.

Best,
Xu
